app/controllers/underpayments/UnderpaymentDetailSummaryController.scala

```diff
@@ -95,15 +95,9 @@ class UnderpaymentDetailSummaryController @Inject() (
         } else {
           cleanupVatAccounting(request).flatMap { updatedRequest =>
             if (request.isRepFlow) {
-              redirectForRepFlow(updatedRequest)
-            } else if (request.checkMode) {
-              Future.successful(Redirect(controllers.cya.routes.CheckYourAnswersController.onLoad()))
-            } else if (request.dutyType == Vat) {
-              Future.successful(Redirect(controllers.underpayments.routes.PostponedVatAccountingController.onLoad()))
-            } else if (request.isOneEntry) {
-              Future.successful(Redirect(controllers.reasons.routes.BoxGuidanceController.onLoad()))
+              redirectForRepFlow(updatedRequest).getOrElse(redirect(updatedRequest))
```

Comment:
If you make redirectForRepFlow call redirect in the "None" case then this would simplify and no need for using Options.